File: lab4/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.google.firebase.auth.FirebaseAuth;

public class MainActivity extends AppCompatActivity {

    private FirebaseAuth auth;
    private TextView welcomeText;
    private TextView usernameText;
    private Button logoutButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        auth = FirebaseAuth.getInstance();
        welcomeText = findViewById(R.id.welcome);
        usernameText = findViewById(R.id.name);
        logoutButton = findViewById(R.id.logout);

        String email = auth.getCurrentUser().getEmail();
        String name = email.split("@")[0];

        usernameText.setText(name);

        logoutButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                logout();
            }
        });
    }

    private void logout() {
        auth.signOut();

        Toast.makeText(MainActivity.this, "Logged out successfully", Toast.LENGTH_SHORT).show();

        Intent intent = new Intent(MainActivity.this, Login.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK); // Xóa mọi activity khác và tạo activity mới
        startActivity(intent);
        finish();
    }
}

File: lab1/lab12/app/src/main/java/com/example/lab1/MainActivity.kt
package com.example.lab1

import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    //khai báo
    var colors = arrayOf("", "", "")
    var myAdapter: ArrayAdapter<String>? = null
    var listView: ListView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        listView = findViewById<ListView>(R.id.listView)
        myAdapter = object :
            ArrayAdapter<String?>(this@MainActivity, android.R.layout.simple_list_item_1, colors) {
            override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
                val view = super.getView(position, convertView, parent)
                if (position == 0) {
                    view.setBackgroundColor(-0xcd00)
                }
                if (position == 1) {
                    view.setBackgroundColor(-0x9a)
                }
                if (position == 2) {
                    view.setBackgroundColor(-0xff7800)
                }
                return view
            }
        }
        listView.setAdapter(myAdapter)
    }
}